### DISCUSSION TITLE
@ApplicationModuleTest in Maven multi-module project

### DISCUSSION THREAD

**Discussion by bheino** [spring-projects/spring-modulith#827] [Q&A]
In our project, we'd like to put each modulith module into a separate maven module. The project structure looks equivalent to the following:

```
multi-module-spring-modulith/
├─ application/
├─ module-a/
```
Packages are structured this way:
```
application - com.mycompany.appname
module-a - com.mycompany.appname.modulea
```
This works fine, modulith detects `module-a` as a separate module when running the application. However, for module-specific tests which use `@ApplicationModuleTest` in `module-a` we get the following error:
```
java.lang.IllegalStateException: Unable to find a @SpringBootConfiguration, you need to use @ContextConfiguration or @SpringBootTest(classes=...) with your test
```
When the class annotated with `@SpringBootApplication` is in the same maven module as the @ApplicationModuleTest, everything works as expected. 

Of course, I could use `@SpringBootTest` instead and configure it via `classes=...` or a `@ContextConfiguration`. By using this approach, I would lose the automatic slicing of the context from `@ApplicationModuleTest` and other benefits (like module structure verification, bootstrap mode etc.).

**TLDR:** Is it possible to use ApplicationModuleTests when each Module is defined in its own maven module? Or is this annotation obsolete in such a scenario?
